typescript/sdk/src/hook/EvmHookModule.ts

```diff
@@ -321,6 +278,79 @@ export class EvmHookModule extends HyperlaneModule<
     return routingHookUpdates;
   }
 
+  protected async updateMutableHook(configs: {
+    current: Exclude<HookConfig, string>;
+    target: Exclude<HookConfig, string>;
+  }): Promise<AnnotatedEV5Transaction[]> {
+    const { current, target } = configs;
+    let updateTxs: AnnotatedEV5Transaction[];
+
+    assert(
+      current.type === target.type,
+      `Mutable hook update requires both hook configs to be of the same type. Expected ${current.type}, got ${target.type}`,
+    );
+    assert(
+      MUTABLE_HOOK_TYPE.includes(current.type),
+      'Expected update config to be of mutable hook type',
+    );
+    // Checking both objects type fields to help typescript narrow the type down correctly
+    switch (true) {
+      case current.type === HookType.INTERCHAIN_GAS_PAYMASTER &&
```

Comment:
nit: switch (true) feels a bit strange here. 

can we use if-statements, or perhaps we can keep the switch and use `current as IgpHookConfig`?

Reply:
I'll change it to if-statements becuase i want to avoid type assertions as much as possible evne if they are safe in this context